### PR TITLE
[ONNX] Fix for batch norm 2D with affine=False

### PR DESCRIPTION
**Pull request by neginraoof**
This is a fix for batch norm 2D with affine=False. 
Repro: https://github.com/pytorch/pytorch/issues/29271
Error is because the output of the unsqueeze op does not have scalar type information. So I moved the references to scalar type after the unsqueeze line.

### CODE REVIEW

**Comment by neginraoof** [comment]
Looking into the CI failure

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -897,6 +894,11 @@ def batch_norm(g, input, weight, bias, running_mean, running_var, training, mome
         bias_value = torch.tensor([0.] * input_sizes[1]).type(
             'torch.' + input.type().scalarType() + 'Tensor')
         bias = g.op("Constant", value_t=bias_value)
+
+    if len(input_sizes) == 2:
+        # batchnorm1d accepts 2d and 3d array, but ONNX only accepts 3d
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py]: ONNX BN seems to allow for 1D tensor, but that seems like a special case. Is that why you do not have a branch for `len(input_sizes) == 1`? Also, we should consider updating the comment here because technically ONNX BN allows 1D.

Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Thanks for mentioning this. I was just thinking about the bug fix so didn't check ONNX spec. Actually looks like this case is supported. Looks like we don't need the branching here. 
It just needs a small fix in Caffe2 spatial batch norm to account for this.
Will update this.

File: caffe2/operators/spatial_batch_norm_op.h [hunk 2]
```diff
@@ -52,7 +52,7 @@ class SpatialBNOp : public Operator<Context> {
     const auto& bias = Input(BIAS);
 
     const int ndim = X.dim();
-    CAFFE_ENFORCE_GE(ndim, 3);
+    CAFFE_ENFORCE_GE(ndim, 2);
```
Comment (spandantiwari) [inline caffe2/operators/spatial_batch_norm_op.h]: I am not sure I understand why this needs to be changed here. Could you explain what was the issue with the Caffe2 op and how this helps.

Comment (neginraoof) [reply caffe2/operators/spatial_batch_norm_op.h]: Based on PyTorch spec, Batch norm 1D can have a shape of (N, C) or (N, C, L). 
Since opset 9, ONNX supports (N, C) shape. However, looks like Caffe2 and the symbolic function have not been updated accordingly.
That's why now we are able to remove the branching for rank==2 in the symbolic,
and we are able to update Caffe 2 with this change.
I'm going to add more tests to cover all cases for ORT and Caffe2.
I'll make sure both set of tests pass.
@houseroad in case this breaks any internal tests for Caffe2, please let us know and we'll revert this.

**Comment by neginraoof** [comment]
Failure does not look related

File: caffe2/operators/spatial_batch_norm_op.h (hunk 2 above)
Comment (neginraoof) [reply caffe2/operators/spatial_batch_norm_op.h]: cc @spandantiwari
Please review the added tests

Comment (spandantiwari) [reply caffe2/operators/spatial_batch_norm_op.h]: Sounds good.

**Review by spandantiwari — APPROVED** [review]
LGTM.

**Comment by neginraoof** [comment]
@houseroad Is the internal build process stuck?

**Comment by spandantiwari** [comment]
@neginraoof  - can you restart this build please?
